File: 2º/PDOO/ProyectoJAVA/src/civitas/C_calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class C_calle extends Casilla{
    
    private TituloPropiedad tituloPropiedad;
    private float importe;
    
    C_calle(TituloPropiedad titulo) {                    // calle
        super(titulo.getNombre());
        tituloPropiedad = titulo;
        importe = titulo.getPrecioCompra();
    }
    
    @Override
    void recibeJugador (int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            super.informe(actual, todos);
            Jugador jugador = todos.get(actual);
            if(!tituloPropiedad.tienePropietario())
                jugador.puedeComprarCasilla();
            else
                tituloPropiedad.tramitarAlquiler(jugador);
        }
    }
    
    TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
    
    @Override
    public String toString() {
        String e = "\n-> Casilla ("+ super.getNombre() + ") " + importe + "$";
        return e;
    }
    
}

File: 2º/PDOO/ProyectoRUBY/lib/c_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "casilla.rb"
require_relative "titulo_propiedad.rb"
require_relative "jugador.rb"

module Civitas
  class C_calle<Casilla
    def initialize(titulo)
      super(titulo.nombre)
      @tituloPropiedad=titulo
      @importe=titulo.precioCompra
    end
    attr_reader :tituloPropiedad, :importe
    def  to_s
      devolver=super()
      devolver=devolver+ "\n\tSu titulo de propiedad es este: #{@tituloPropiedad.to_s}";
      devolver+="\n}\n"
      return devolver;
    end  
    def recibeJugador(iactual,todos)
        if(jugadorCorrecto(iactual,todos)==true)then
          informe(iactual,todos)
          jugador=todos[iactual]
          if(@tituloPropiedad.tienePropietario==false)then
            jugador.puedeComprarCasilla
          else  
            @tituloPropiedad.tramitarAlquiler(jugador)
          end
        end
      end
  end
end
